src/react-web3/Web3Provider.js

```diff
@@ -9,30 +10,76 @@ class Web3Provider extends Component {
     super(props, context)
 
     this.state = {
-      selectedAccount: null
+      selectedAccount: null,
+      web3: null,
+      render: false
     }
 
-    this.web3 = null
     this.interval = null
-
   }
+
   render() {
     const { web3UnavailableScreen: Web3UnavailableScreen } = this.props
+    let renderContainer = false
 
-    if (window.web3) {
-      if (!this.web3) {
-        this.web3 = new Web3(window.web3.currentProvider);
-        this.fetchAccounts()
+    if (this.state.render) {
+      if (this.state.web3) {
+        renderContainer = this.props.children
+      } else {
+        renderContainer =  <Web3UnavailableScreen/>
       }
-
-      return this.props.children
+    } else {
+      renderContainer = <Loader show={true}></Loader>
     }
+    return ( renderContainer )
 
-    return <Web3UnavailableScreen />
   }
 
   componentDidMount() {
-    this.initPoll();
+    this.checkWeb3()
+    this.initPoll()
+
+    // Wait 1 second to render
+    setTimeout(function() {
+      this.setState({render: true})
+    }.bind(this), 1500)
+  }
+
+  checkWeb3() {
+    const setWeb3 = () => {
+      window.web3 = new Web3(window.web3.currentProvider)
+      this.setState({
+        web3: window.web3
+      })
+      this.fetchAccounts()
```

Comment:
Here `fetchAccounts` is being called, but it's called aswell inside the `interval` defined in the line `91`, which is called after calling `checkWeb3`, in the line `40`